includes/Core/Util/Entity.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Entity
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use Google\Site_Kit\Context;
+
+/**
+ * Class representing an entity.
+ *
+ * An entity in Site Kit terminology is based on a canonical URL, i.e. every
+ * canonical URL has an associated entity.
```

Comment:
Not sure I understand this entirely. Does that mean that all site URLs would have an entity?

Would 404, feeds, or REST endpoints not be exceptions? What about the login or registration pages?

Reply:
I guess frontend canonical URLs. The 404 is a special case, you could argue that even that is a specific "endpoint" in WordPress. I'll clarify the comment.